app-ng/src/make-qvm.lisp

```diff
@@ -0,0 +1,57 @@
+;;;; src/make-qvm.lisp
+;;;;
+;;;; Author: appleby
+
+(in-package #:qvm-app-ng)
+
+(defun make-requested-allocation-descriptor (allocation-method length)
+  (ecase allocation-method
+    (native  (make-instance 'qvm:lisp-allocation :length length))
+    (foreign (make-instance 'qvm:c-allocation :length length))))
+
+(defgeneric make-requested-qvm (simulation-method allocation-method num-qubits
+                                gate-noise measurement-noise)
+  (:documentation "Create and return a QVM instance of the requested type."))
+
+(defmethod make-requested-qvm ((simulation-method (eql 'pure-state))
+                               allocation-method
+                               num-qubits
+                               (gate-noise null)
+                               (measurement-noise null))
+  (qvm:make-qvm num-qubits :allocation (make-requested-allocation-descriptor
+                                        allocation-method
+                                        (expt 2 num-qubits))))
+
+(defmethod make-requested-qvm ((simulation-method (eql 'pure-state))
+                               allocation-method
+                               num-qubits
+                               gate-noise
+                               measurement-noise)
+  (let ((gate-noise (or gate-noise (load-time-value '(0.0 0.0 0.0))))
+        (measurement-noise (or measurement-noise (load-time-value '(0.0 0.0 0.0)))))
+    ;; TODO(appleby): Perhaps we should add QVM:MAKE-DEPOLARIZING-QVM similar to QVM:MAKE-QVM and
+    ;; QVM:MAKE-DENSITY-QVM. Along the way, it would be good to homogenize the interfaces and also
+    ;; add QVM:MAKE-NOISY-QVM for completeness.
```

Comment:
Assuming this TODO is still relevant, I can make sure to include `make-xx-qvm` functions for all the different noisy qvms I recently added

Reply:
That would be awesome, thank you.  I was actually reminded of this TODO when I was looking over your PR this morning :).  For sure it would be nice from qvm-app's perspective to have a unified way to ask qvm: "give me a qvm with this many qubits and these characteristics (state rep, noise, allocation method, etc)"

Reply:
I went ahead and filed an issue to track this #223. Removing this TODO for now.